{
  if (Callback.Checked.class.isAssignableFrom(this.callback.getClass())) {
    try {
      if ((failure instanceof ExecutionException) && failure.getCause() != null) {
        failure=failure.getCause();
      }
      EventRecord.caller(this.callback.getClass(),EventType.CALLBACK,"fireException(" + failure.getClass().getSimpleName() + ")").trace();
      ((Checked)this.callback).fireException(failure);
    }
 catch (    Throwable t) {
      LOG.error(t,t);
    }
  }
}
